docs/book/developer-guide/steps-pipelines/caching.md

```diff
@@ -0,0 +1,181 @@
+---
+description: How does automatic caching work in ZenML.
+---
+
+Machine learning pipelines are rerun many times over throughout their development lifecycle. Prototyping is often a 
```

Comment:
I would in passing mention that materializers are the things that version data for ZenML